jax/_src/lax/control_flow/for_loop.py

```diff
@@ -0,0 +1,262 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for the `for_loop` primitive."""
+from functools import partial
+
+from typing import Any, List, Tuple
+
+from jax import core
+from jax.interpreters import ad
+from jax._src import ad_util
+from jax._src import pretty_printer as pp
+
+
+## Helpful type aliases
+Ref = Array = Any
+
+## State effect
+
+State = object()
```

Comment:
```suggestion
class StateEffect: pass
State = StateEffect
```

Leads to nicer pdb printing if/when necessary!